### ISSUE TITLE
Variable wrongly labeled as unused

### ISSUE THREAD

**Issue by Tharit** [sveltejs/eslint-plugin-svelte3#64]
It seems like there is some issue with variables being wrongly labeled as "unused" when their value is modified in an inline handler on a DOM node.

Minimal Example:
```
<script>
    let test = 0;
    function action() {
        console.log(test);
    }
</script>

<div>
    <div on:click={()=>test=1}></div>
    <div on:click={action}></div>)
</div>
```

Output:
Error: 'test' is assigned a value but never used

Used versions:
```
"devDependencies": {
    "eslint": "^7.2.0",
    "eslint-plugin-jsdoc": "^27.0.7",
    "eslint-plugin-svelte3": "^2.7.3",
    "svelte": "^3.23.2"
  }
```

Seems like no matter what happens in the <script> part, 'test' is always detected as unused.

**Comment by MacFJA** [comment]
I did some quick test, the issue appear with ESLint v7.2.0

From the [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md), and be reverting the changes, it's this commit that change the behavior: https://github.com/eslint/eslint/commit/426088c966dc79dc338b33100f3adf827b147d69

**Comment by Conduitry** [comment]
Sorry about the long delay on actually merging and releasing the fix. This should be finally fixed in v3.0.0. (The major version bump is because Node 8 is no longer officially supported.)
